clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2774,6 +2774,9 @@ void handleResponse(AbstractResponse abstractResponse) {
                             context.node().get(),
                             authorizedOperations);
                     context.future().complete(consumerGroupDescription);
+                } else {
+                    context.future().completeExceptionally(new IllegalArgumentException(
```

Comment:
What are your thoughts on having `InvalidGroupIdException` here?
In all regards the KafkaFuture will wrap it in an `ExecutionException` - I'm just wondering if user code could be checking whether the cause is a `KafkaException`.

Reply:
I thought about this and I felt like `InvalidGroupIdException` was inappropriate because the group is is valid but the type of the group is not accepted. Thus, `IllegalArgumentException` sounds better to me in this case. Does it make sense?

Reply:
Yeah it's not straightforward. Let's see what others think

Reply:
I'm ok with `IllegalArgumentException`. I think `InvalidGroupIdException` is typically used to indicate a groupId which is structurally invalid in some way (e.g. null).